File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/PostDestinationExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class PostDestinationExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey.Add("Authorization", "YOUR_API_KEY");

        var destinationPost = new DestinationPost(
            kind: DestinationPost.KindEnum.GooglePubsub
        );

        try
        {
            var response = new DataExportDestinationsApi(config).PostDestination(
                projectKey: "projectKey_string",
                environmentKey: "environmentKey_string",
                destinationPost: destinationPost
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling DataExportDestinationsApi#PostDestination: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-axios/src/PostDestinationExample.ts
import axios, { AxiosError } from "axios";
import * as api from "launchdarkly_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

const destinationPost: api.DestinationPost = {
  kind: api.DestinationPostKindEnum.GooglePubsub,
};

new api.DataExportDestinationsApi(configuration).postDestination(
  "projectKey_string", // projectKey
  "environmentKey_string", // environmentKey
  destinationPost,
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling DataExportDestinationsApi#postDestination:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
